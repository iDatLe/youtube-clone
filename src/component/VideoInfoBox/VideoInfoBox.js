import React, { useState } from 'react';
import './VideoInfoBox.scss';
import { Image, Button, Divider } from 'semantic-ui-react';

export function VideoInfoBox() {
    const [collapsed, setCollapsed] = useState(false);

    const onToggleCollapseButtonClick = () => {
        setCollapsed(!collapsed)
    }

    let descriptionTextClass = 'collapsed';
    let buttonTitle = 'Show More';
    if (!collapsed) {
        descriptionTextClass = 'expanded';
        buttonTitle = 'Show Less';
    }

    return (
        <>
            <div className='video-info-box'>
                <Image className='channel-image' src='http://via.placeholder.com/48x48' circular/>
                <div className="video-info">
                    <div className='channel-name'>Channel Name</div>
                    <div className='video-publication-date'>Thu 24, 2017</div>
                </div>
                <Button className='subscribe' color='youtube'>91.5K Subscribe</Button>
                <div className="video-description">
                    <div className={descriptionTextClass}>
                        {/*descriptionParagraphs*/}
                        <p>paragraphs</p>
                        <p>paragraphs</p>
                        <p>paragraphs</p>
                        <p>paragraphs</p>
                        <p>paragraphs</p>
                        <p>paragraphs</p>
                        <p>paragraphs</p>
                    </div>
                    <Button compact onClick={onToggleCollapseButtonClick}>{buttonTitle}</Button>
                </div>
            </div>
            <Divider />
        </>
    )
}

