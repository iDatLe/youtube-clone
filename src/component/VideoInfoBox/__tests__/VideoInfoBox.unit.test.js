import React from 'react';
import { shallow } from 'enzyme';
import { VideoInfoBox } from '../VideoInfoBox';

describe('VideoInfoBox render', () => {
    test('render collapsed', () => {
        const wrapper = shallow(
            <VideoInfoBox />
        );
        expect(wrapper).toMatchSnapshot();
    })

    test('render expanded', () => {
        const wrapper = shallow(
            <VideoInfoBox />
        );
        // wrapper.setState({collapse: false})
        expect(wrapper).toMatchSnapshot();
    });
});