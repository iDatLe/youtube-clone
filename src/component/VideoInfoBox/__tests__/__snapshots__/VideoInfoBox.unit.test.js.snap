// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfoBox render render collapsed 1`] = `
<Fragment>
  <div
    className="video-info-box"
  >
    <Image
      as="img"
      circular={true}
      className="channel-image"
      src="http://via.placeholder.com/48x48"
      ui={true}
    />
    <div
      className="video-info"
    >
      <div
        className="channel-name"
      >
        Channel Name
      </div>
      <div
        className="video-publication-date"
      >
        Thu 24, 2017
      </div>
    </div>
    <Button
      as="button"
      className="subscribe"
      color="youtube"
    >
      91.5K Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
      </div>
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
      >
        Show Less
      </Button>
    </div>
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoInfoBox render render expanded 1`] = `
<Fragment>
  <div
    className="video-info-box"
  >
    <Image
      as="img"
      circular={true}
      className="channel-image"
      src="http://via.placeholder.com/48x48"
      ui={true}
    />
    <div
      className="video-info"
    >
      <div
        className="channel-name"
      >
        Channel Name
      </div>
      <div
        className="video-publication-date"
      >
        Thu 24, 2017
      </div>
    </div>
    <Button
      as="button"
      className="subscribe"
      color="youtube"
    >
      91.5K Subscribe
    </Button>
    <div
      className="video-description"
    >
      <div
        className="expanded"
      >
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
        <p>
          paragraphs
        </p>
      </div>
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
      >
        Show Less
      </Button>
    </div>
  </div>
  <Divider />
</Fragment>
`;
